### ISSUE TITLE
Why the versions of OS and gcc in build_python.yml are old?

### ISSUE THREAD

**Issue by zaikunzhang** [libprima/prima#199]
Hi @nbelakovski ,

Is it intended that we use older versions of OS (instead of latest) and `gcc` 8 (for macOS, very old) in 

https://github.com/libprima/prima/blob/main/.github/workflows/build_python.yml

In addition, we need to cover macOS ARM64, currently available under GitHub Actions with `macos-14` runners.

Thank you.

Zaikun

**Comment by nbelakovski** [comment]
Some of the code there was copied from https://github.com/scipy/scipy/blob/main/.github/workflows/wheels.yml, particularly the rtools part, which installed gcc8, and hence why I used gcc8 elsewhere.

As for the OS versions, it looks like I copied from the cibuildwheels docs, specifically [this one](https://cibuildwheel.pypa.io/en/stable/setup/#github-actions), which currently uses `-latest` for Windows/Linux and more recent versions for MacOS, but a couple months ago, when I wrote this file, it was showing the [older versions](https://github.com/pypa/cibuildwheel/blob/ce3fb7832089eb3e723a0a99cab7f3eaccf074fd/docs/setup.md?plain=1#L181), so I decided to just copy in case it wasn't yet working on that latest builds.

If the versions can be safely updated I see no reason why not.
